components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,331 @@
+  initialLayout = true;
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+  [super viewWillAppear:animated];
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  self.mdc_bottomSheetPresentationController.delegate = self;
+#pragma clang diagnostic pop
+
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap =
+      _transitionController.dismissOnBackgroundTap;
+  [self.view layoutIfNeeded];
+}
+
+- (BOOL)accessibilityPerformEscape {
+  if (!self.dismissOnBackgroundTap) {
+    return NO;
+  }
+  [self dismissViewControllerAnimated:YES completion:nil];
+  return YES;
+}
+
+- (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentContainer>)container {
+  [super preferredContentSizeDidChangeForChildContentContainer:container];
+  [self.presentationController
+      preferredContentSizeDidChangeForChildContentContainer:self];
+}
+
+- (UIScrollView *)trackingScrollView {
+  return _transitionController.trackingScrollView;
+}
+
+- (void)setTrackingScrollView:(UIScrollView *)trackingScrollView {
+  _transitionController.trackingScrollView = trackingScrollView;
+}
+
+- (BOOL)dismissOnBackgroundTap {
+  return _transitionController.dismissOnBackgroundTap;
+}
+
+-(void)setDismissOnBackgroundTap:(BOOL)dismissOnBackgroundTap {
+  _transitionController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+}
+
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:
+    (id<UIViewControllerTransitionCoordinator>)coordinator {
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+  CGRect frame = self.view.frame;
+  frame.size = size;
+  self.view.frame = frame;
+  CGRect headerFrame = self.header.frame;
+  headerFrame.size.width = size.width;
+  self.header.frame = headerFrame;
+  [self.header setNeedsLayout];
+  [self.header layoutIfNeeded];
+
+  CGFloat height = CGRectGetHeight(self.header.frame) + [_tableView tableHeightForWidth:size.width];
+  CGSize updatedSize = CGSizeMake(size.width, height);
+  self.preferredContentSize = updatedSize;
+}
+
+-(void)setTransitioningDelegate:(id<UIViewControllerTransitioningDelegate>)transitioningDelegate {
+  NSAssert(NO, @"MDCActionSheetController.transitionDelegate cannot be changed");
+  return;
+}
+
+-(void)setModalPresentationStyle:(__unused UIModalPresentationStyle)modalPresentationStyle {
+  NSAssert(NO, @"MDCActionSheetController.modalPresentationStyle cannot be changed.");
+  return;
+}
+
+- (MDCActionSheetHeaderView *)headerView {
+  CGRect headerRect = CGRectZero;
+  headerRect.size.width = CGRectGetWidth(self.view.frame);
+  MDCActionSheetHeaderView *header = [[MDCActionSheetHeaderView alloc] initWithFrame:headerRect];
+  header.title = _actionSheetTitle;
+  header.message = _message;
+  header.mdc_adjustsFontForContentSizeCategory = self.mdc_adjustsFontForContentSizeCategory;
+  return header;
```

Comment:
Consider moving this logic into viewDidLoad so that it can't be accidentally invoked elsewhere in your view controller (you only ever want one header I'm assuming).